hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/list/TestS3AReadRestoredGlacierObjects.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.fs.s3a.list;
+
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.s3a.AbstractS3AMockTest;
+import org.apache.hadoop.fs.s3a.S3ObjectStorageClassFilter;
+import org.apache.hadoop.fs.s3a.Listing;
+import org.apache.hadoop.fs.s3a.Listing.GlacierStatusAcceptor;
+import org.junit.Assert;
+import org.junit.Test;
+import software.amazon.awssdk.services.s3.model.ObjectStorageClass;
+import software.amazon.awssdk.services.s3.model.RestoreStatus;
+import software.amazon.awssdk.services.s3.model.S3Object;
+
+public class TestS3AReadRestoredGlacierObjects extends AbstractS3AMockTest {
+
+  String s3Path = "s3://test_bucket/path";
+
+  @Test
+  public void testIgnoreGlacierObject() {
+    S3ObjectStorageClassFilter restoredObjectsType = S3ObjectStorageClassFilter.SKIP_ALL_GLACIER;
+
+    Listing.GlacierStatusAcceptor acceptor = new GlacierStatusAcceptor(restoredObjectsType);
+
+    boolean result = acceptor.accept(new Path(s3Path),
+        getS3ObjectWithStorageClassAndRestoreStatus(
+            s3Path,
+            RestoreStatus.builder().isRestoreInProgress(false).build(),
+            ObjectStorageClass.GLACIER));
+
+    Assert.assertFalse(result);
```

Comment:
i require a meaningful error message on all failures. ideally using assertj whose describedAs() class can do string formatting.

```
o = getS3ObjectWithStorageClassAndRestoreStatus(...)
Assertions.assertThat(acceptor.accept(...))
 .describedAs("accept %s", o)
 .isFalse()
```

just imagine you've seen a test failure: what information would you want in the assertion message to begin debugging it?